airbyte-cdk/java/airbyte-cdk/core/src/testFixtures/kotlin/io/airbyte/cdk/extensions/LoggingInvocationInterceptor.kt

```diff
@@ -36,39 +36,45 @@ class LoggingInvocationInterceptor : InvocationInterceptor {
     private class LoggingInvocationInterceptorHandler : InvocationHandler {
         @Throws(Throwable::class)
         override fun invoke(proxy: Any, method: Method, args: Array<Any>): Any? {
-            if (
+            val methodName = method.name
+            val invocationContextClass: Class<*> =
+                when (methodName) {
+                    "interceptDynamicTest" -> DynamicTestInvocationContext::class.java
+                    else -> ReflectiveInvocationContext::class.java
+                }
+            try {
                 LoggingInvocationInterceptor::class
                     .java
                     .getDeclaredMethod(
                         method.name,
                         InvocationInterceptor.Invocation::class.java,
-                        ReflectiveInvocationContext::class.java,
+                        invocationContextClass,
                         ExtensionContext::class.java
-                    ) == null
-            ) {
+                    )
+            } catch (_: NoSuchMethodException) {
```

Comment:
When the method isn't found, `getDeclaredMethod` doesn't return null, instead it always throws `NoSuchMethodException`.